dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -52,6 +55,20 @@ def show
          locale: locale) if @script_level.level.finishable?
   end
 
+  protected
+
+  # Prevent caching for the current page, unless the Gatekeeper configuration for 'script' specifies
+  # that it is publicly cachable, in which case the max-age and s-max-age headers are set based
+  # the 'public-max-age' DCDO configuration value.
+  def prevent_caching_unless_enabled_for_script(script)
```

Comment:
hm, the name of this method is a bit double-negativey... maybe something like configure_caching? (especially since it also does some configuration stuff even if it's not preventing caching)